docs/docs/kds-configuration.md

```diff
@@ -1,47 +1,39 @@
-# Configuring Conclave to use a key from a Key Derivation Service
-Conclave enclaves are able to use a key from a Key Derivation Service (KDS) for
-encrypting data that will be persisted outside the enclave. Using a key from a
-KDS allows the encrypted data that is persisted by an enclave to be accessed by
-the enclave even if it is moved to another physical machine. Without the KDS,
-moving an enclave to a different physical machine renders any encrypted
-persisted data unreadable, effectively cryptographically erasing the data.
+# Configuring Conclave for use with the KDS
+This page covers how to configure your enclave project to use KDS keys.
+More detailed information about the architecture of the KDS can be found [here](kds-detail.md).
 
-R3 provides an instance of a Conclave Key Derivation Service (KDS) as part of
-the Conclave Platform that can be used by any Conclave enclave to obtain stable
-keys for persisting data regardless of what physical system the enclave is
-running on.
+## What is the KDS?
+By default, Conclave persists encrypted data outside the enclave using keys which are specific to the CPU which the enclave is running on.
+This causes problems for cloud applications where virtual machines may be re-provisioned and moved to different hardware.
 
-This page describes how to configure your enclave project to use a KDS key. For
-a [more detailed description of how this works and the reasons behind using
-a KDS, please take a look here](kds-detail.md).
+In addition to CPU based keys for storage, Conclave makes use of random session keys for communication.
+This presents additional problems for cloud applications which involve horizontal scaling, because the session key will be different for each enclave in the cluster.
 
+The Conclave Key Derivation Service (KDS) is an enclave based service which solves these problems by providing Conclave enclaves a means to securely derive stable keys for both storage and communication.
+R3 provides a publicly accessible instance of the KDS as part of the Conclave Platform which may be used by any Conclave based enclave.
 
-## Configuring the KDS URL
-For an enclave to obtain a key from the KDS, the URL of the KDS and the
-specification of the key to obtain must both be configured by the developer.
+!!!Note
+The URL for the R3 public KDS is `https://kds.dev.conclave.cloud`.
```

Comment:
This is not the correct syntax. You can see how it's rendered by looking at the docsite output from TC: https://r3-conclave.artifacts.teamcity.com/presignedTokenAuth/fbffc162-7a85-43ef-8c49-3d6cc71ac1a9/repository/download/ConclaveCore_SdkBuild/3619:id/docs.zip!/kds-configuration.html

Reply:
Yeah, intellij keeps removing the indentation here. Not sure why. It's been fixed.